### PR TITLE
Add case-conversion functions (#1376)

### PR DESCRIPTION
**Pull request by gVirtu**
As per #1376, adds a thin wrapper to [heck](https://github.com/withoutboats/heck)'s case conversion utilities, along with straightforward tests to each one.

Hopefully it ends up being useful! :)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1145,12 +1145,24 @@ The executable is at: /bin/just
 
 - `capitalize(s)`<sup>master</sup> - Convert first character of `s` to uppercase and the rest to lowercase.
 
+- `kebabcase(s)`<sup>master</sup> - Convert `s` to kebab case.
```
Comment (casey) [inline README.md]:
```suggestion
- `kebabcase(s)`<sup>master</sup> - Convert `s` to `kebab-case`.
```

File: README.md [hunk 2]
```diff
@@ -1145,12 +1145,24 @@ The executable is at: /bin/just
...
+
+- `lowercamelcase(s)`<sup>master</sup> - Convert `s` to lower camel case. (also known simply as camel case)
```
Comment (casey) [inline README.md]:
```suggestion
- `lowercamelcase(s)`<sup>master</sup> - Convert `s` to `lowerCamelCase`.
```

File: README.md [hunk 3]
```diff
@@ -1145,12 +1145,24 @@ The executable is at: /bin/just
...
+
 - `lowercase(s)` - Convert `s` to lowercase.
 
 - `quote(s)` - Replace all single quotes with `'\''` and prepend and append single quotes to `s`. This is sufficient to escape special characters for many shells, including most Bourne shell descendants.
 
 - `replace(s, from, to)` - Replace all occurrences of `from` in `s` to `to`.
 
+- `shoutykebabcase(s)`<sup>master</sup> - Convert `s` to kebab case in all capitals.
```
Comment (casey) [inline README.md]:
```suggestion
- `shoutykebabcase(s)`<sup>master</sup> - Convert `s` to `SHOUTY-KEBAB-CASE`.

File: README.md [hunk 4]
```diff
@@ -1145,12 +1145,24 @@ The executable is at: /bin/just
...
+
+- `shoutysnakecase(s)`<sup>master</sup> - Convert `s` to snake case in all capitals.
```
Comment (casey) [inline README.md]:
```suggestion
- `shoutysnakecase(s)`<sup>master</sup> - Convert `s` to `SHOUTY_SNAKE_CASE`.
```

File: README.md [hunk 5]
```diff
@@ -1145,12 +1145,24 @@ The executable is at: /bin/just
...
+
+- `snakecase(s)`<sup>master</sup> - Convert `s` to snake case.
```
Comment (casey) [inline README.md]:
```suggestion
- `snakecase(s)`<sup>master</sup> - Convert `s` to `snake_case`.
```

File: README.md [hunk 6]
```diff
@@ -1145,12 +1145,24 @@ The executable is at: /bin/just
...
+
+- `titlecase(s)`<sup>master</sup> - Convert `s` to title case.
```
Comment (casey) [inline README.md]:
```suggestion
- `titlecase(s)`<sup>master</sup> - Convert `s` to `Title Case`.
```

File: README.md [hunk 7]
```diff
@@ -1167,6 +1179,8 @@ The executable is at: /bin/just
 
 - `uppercase(s)` - Convert `s` to uppercase.
 
+- `uppercamelcase(s)`<sup>master</sup> - Convert `s` to upper camel case. (also known as Pascal case)
```
Comment (casey) [inline README.md]:
```suggestion
- `uppercamelcase(s)`<sup>master</sup> - Convert `s` to `UpperCamelCase`.
```

**Review by casey — CHANGES_REQUESTED** [review]
Very nice! To make it super obvious what the different functions do, let's use the destination case in the description of each function.

**Review by casey — APPROVED** [review]
I went ahead and committed the suggested changes to the readme.

**Comment by casey** [comment]
FYI, since `just` is a binary crate, changes to `Cargo.lock` should be committed.

Merged, thank you for the PR! I'm sure people will wind up using this. Case conversion is common enough, and case conversion in the shell is pretty miserable.

**Comment by gVirtu** [comment]
@casey thank you!! I'd signed off for the day so appreciate the help.

I wasn't sure about the `Cargo.lock` change (newbie to the rust ecosystem :smiley: ) since `heck` ended up being the only one with a version number so I suspected it was a Rust version mismatch on my end.

Thank you once more and congratulations on maintaining the project, it's organized very nicely, easy to set up and well tested. I'm starring it myself!

**Comment by casey** [comment]
You bet, and thank you for the kind words! Contributions like yours make it fun and motivating to work on in the long term, so thank you!

**Pull request merged**